bookkeeper-server/src/main/java/org/apache/bookkeeper/client/BookKeeperServerHealthInfo.java

```diff
@@ -0,0 +1,65 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.apache.bookkeeper.client;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.bookkeeper.net.BookieSocketAddress;
+
+/**
+ * This class encompasses heuristics used to determine the health of a Bookkeeper server for read
+ * ordering. Currently we use the history of failed entries and length of the pending requests queue.
+ */
+public class BookKeeperServerHealthInfo {
```

Comment:
this class provides info about multiple bookies.
what about "BookiesHealthInfo" ?

Reply:
@eolivelli If naming is not a big concern, let's skip it or rename it later, since it is a port back from twitter branch.

Reply:
Ok I didn't know it is a port. Maybe it was not clear from the description.
+1

Reply:
Ok I didn't know it is a port. Maybe it was not clear from the description.
+1

Reply:
I could make this change - it wasn't a clean port so I had to reimplement a lot of stuff anyways.